### ISSUE TITLE
Batchloading BelongsTo repeats last retrieved model

### ISSUE THREAD

**Issue by MissaelAnda** [nuwave/lighthouse#2063]
When retrieving a nested (3 relations deep) belongsTo relation the last retrived model of that relation gets placed for all belongsTo relations.

This is solved by desabling batchloading.

The belongs to relations is accessed via two has many relations

![imagen](https://user-images.githubusercontent.com/31199350/153652592-bc622251-63c8-455a-862a-1a591653e1b0.png)

Lighthouse Version: 5.39.2

**Comment by spawnia** [comment]
Please try and add a failing test case to https://github.com/nuwave/lighthouse/blob/master/tests/Integration/Schema/Directives/BelongsToDirectiveTest.php

**Comment by alberthaff** [comment]
I just spent a couple of hours debugging the same error.

In my case, the problem turned out to be, that I'm using a model which has a string as primary key. I had not added following to the model:
```php
public $incrementing = false;
```
As soon as I added this, it solved the problem.
From your example @MissaelAnda, it looks like the child model might be using a string as primary key as well.
But it might just be caused by `modifier.id` being typed as a string in your schema?

I'm loading the parent model from a MySQL view, and because of this I had not given the model's configuration much thought, since I won't ever be doing any writes to the database with the given model.
I assume the fact that I'm using a view instead of a table doesn't really matter for this scenario, but I haven't tested.
